lib/routes/user/home/wallet_dashboard.dart

```diff
@@ -18,58 +19,116 @@ class WalletDashboard extends StatelessWidget {
-              child: _accountModel != null && !_accountModel.initial
-                  ? Text("Balance", style: theme.subtitle.copyWith(color: theme.subtitle.color.withOpacity( pow(1 - _offsetFactor, 8))))
-                  : SizedBox(),
-            )),
-        Positioned(
-            top: 30 - BALANCE_OFFSET_TRANSITION * _offsetFactor,
-            child: Center(
-              child: _accountModel != null && !_accountModel.initial
-                  ? FlatButton(
-                  onPressed: () {
-                    var nextCurrencyIndex = (Currency.currencies.indexOf(_accountModel.currency) + 1) % Currency.currencies.length;
-                    _onCurrencyChange(Currency.currencies[nextCurrencyIndex]);
-                  },
-                  child: Text(_accountModel.currency.format(_accountModel.balance, fixedDecimals: false),
-                      style: theme.headline.copyWith(fontSize: startHeaderSize - (startHeaderSize - endHeaderFontSize) * _offsetFactor)))
-                  : SizedBox(),
+            showProgressBar ? Positioned(top: 0.0, child: StatusIndicator(widget._accountModel)) : SizedBox(),
+            Positioned(
+                top: 10.0,
+                child: Center(
+                  child: widget._accountModel != null && !widget._accountModel.initial
+                      ? Text("Balance",
+                          style: theme.subtitle.copyWith(color: theme.subtitle.color.withOpacity(pow(1 - widget._offsetFactor, 8))))
+                      : SizedBox(),
+                )),
+            Positioned(
+              top: 30 - BALANCE_OFFSET_TRANSITION * widget._offsetFactor,
+              child: Center(
+                child: widget._accountModel != null && !widget._accountModel.initial
+                    ? FlatButton(
+                        onPressed: () {
+                          var nextCurrencyIndex =
+                              (Currency.currencies.indexOf(widget._accountModel.currency) + 1) % Currency.currencies.length;
+                          widget._onCurrencyChange(Currency.currencies[nextCurrencyIndex]);
+                        },
+                        child: _showFiatCurrency
+                            ? Text("${widget._accountModel.fiatCurrency.symbol}${_convertedBalance.toStringAsFixed(2)}",
+                                style: theme.headline
+                                    .copyWith(fontSize: startHeaderSize - (startHeaderSize - endHeaderFontSize) * widget._offsetFactor))
+                            : Text(widget._accountModel.currency.format(widget._accountModel.balance, fixedDecimals: false),
+                                style: theme.headline
+                                    .copyWith(fontSize: startHeaderSize - (startHeaderSize - endHeaderFontSize) * widget._offsetFactor)))
+                    : SizedBox(),
               ),
             )
-      ],
-    );
+          ],
+        ),
+        behavior: HitTestBehavior.translucent,
```

Comment:
Can't we use a gesture here instead of the pointerup/pointerdown logic? flutter support "long press" and "tap" gestures.